clients/src/main/java/org/apache/kafka/common/utils/LambdaUtils.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.utils;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Lambda helpers.
+ */
+@FunctionalInterface
+public interface LambdaUtils {
+    /**
+     * Run some code, possibly throw some exceptions.
+     *
+     * @throws Exception
+     */
+    void run() throws Exception;
+
+    /**
+     * Provide an idempotent instance of the supplied code - ensure that the supplied code gets run only once, no
+     * matter how many times .run() is called.
+     */
+    static Runnable idempotent(final Runnable code) {
```

Comment:
Is this one used anywhere? I can't find it.

Reply:
No, it's not used anywhere, I added it as a helper for future. But now I have removed this file completely and moved only the essential functions to `Utils.java`